tree/10_ncf_internals/configuration.cf

```diff
@@ -47,18 +45,21 @@ bundle agent configuration
       "enabled_loggers"            slist => splitstring("${ncf_config[loggers][1]}",",","10");
       "enabled_abort_handlers"     slist => splitstring("${ncf_config[abort_handlers][1]}",",","10");
       "cfengine_port"              string => "${ncf_config[cfengine_port][1]}";
+      "flag_file"                  string => "${ncf_config[flag_file][1]}";
 
       # Add a default empty list to allow skipping those handlers
-    pass3.enabled_loggers_empty::
+    pass2.enabled_loggers_empty::
       "enabled_loggers"            slist => {};
-    pass3.enabled_abort_handlers_empty::
+    pass2.enabled_abort_handlers_empty::
       "enabled_abort_handlers"     slist => {};
-    pass3.!enabled_loggers_defined::
+    pass2.!enabled_loggers_defined::
       "enabled_loggers"            slist => { "_log_default" };
-    pass3.!enabled_abort_handlers_defined::
+    pass2.!enabled_abort_handlers_defined::
       "enabled_abort_handlers"     slist => { "_abort_default" };
-    pass3.!cfengine_port_defined::
+    pass2.!cfengine_port_defined::
       "cfengine_port"              string => "5308";
+    pass2.!flag_file_defined::
+      "flag_file"                  string => "/var/rudder/agent-data/flags.json";
```

Comment:
we need something to create this folder

Reply:
Indeed, this should be part of the system techniques I think